instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/servlet/AppServerBridge.java

```diff
@@ -42,42 +50,30 @@ public static Context init(Context ctx, boolean shouldRecordException) {
     return ctx.with(AppServerBridge.CONTEXT_KEY, new AppServerBridge(shouldRecordException));
   }
 
-  private final AtomicBoolean servletUpdatedServerSpanName = new AtomicBoolean(false);
-  private final AtomicBoolean servletShouldRecordException;
+  private final boolean servletShouldRecordException;
+
+  private volatile int servletUpdatedServerSpanName = FALSE;
 
   private AppServerBridge(boolean shouldRecordException) {
-    servletShouldRecordException = new AtomicBoolean(shouldRecordException);
+    servletShouldRecordException = shouldRecordException;
   }
 
   /**
-   * Returns true, if servlet integration should update server span name. After server span name has
-   * been updated with <code>setServletUpdatedServerSpanName</code> this method will return <code>
-   * false</code>.
+   * Returns true, if servlet integration has not already updated the server span name. Subsequent
+   * invocations will return {@code false}. This is meant to be used in a compare-and-set fashion.
    *
    * @param ctx server context
-   * @return <code>true</code>, if the server span name should be updated by servlet integration, or
-   *     <code>false</code> otherwise.
+   * @return <code>true</code>, if the server span name had not already been updated by servlet
+   *     integration, or <code>false</code> otherwise.
    */
-  public static boolean shouldUpdateServerSpanName(Context ctx) {
+  public static boolean setUpdatedServerSpanName(Context ctx) {
     AppServerBridge appServerBridge = ctx.get(AppServerBridge.CONTEXT_KEY);
     if (appServerBridge != null) {
-      return !appServerBridge.servletUpdatedServerSpanName.get();
+      return servlerUpdatedServerSpanNameUpdater.compareAndSet(appServerBridge, FALSE, TRUE);
```

Comment:
This previously wasn't using CAS but I presume it wanted to, or otherwise I could just change to a simple `volatile`.